server/web/auth/__init__.py

```diff
@@ -39,10 +40,49 @@ def signup():
                     email=form.email.data,
                     password=form.password.data)
         db.session.add(user)
+        token = user.generate_token()
+        if current_app.config['DEBUG']:
+            print(url_for('auth.confirm', token=token, _external=True))
+        send_email(user.email, 'Confirm Your Account',
+                   'auth/email/confirm', user=user, token=token)
+        flash('You have been registered. A confirmation email is sent to your email address. \
+               You have 24 hours to verify your account.')
         db.session.commit()
-        flash('You have been registered')
         login_user(user)
-        return redirect(url_for('main.index'))  # TODO: Should redirect to dash index
     return render_template('auth/signup.html.j2', form=form)
 
 
+@auth.route('/confirm/<token>')
+@login_required
+def confirm(token):
+    if current_user.verified:
+        return redirect(url_for('main.index'))
+    if current_user.confirm(token):
+        db.session.commit()
+        flash('You have verified your account!')
+    elif current_user.confirm(token) == 'expire':
+        flash('The confirmation link is expired')
+    else:
+        flash('The confirmation link is invalid')
+        return redirect(url_for('main.404'))
+    return redirect(url_for('main.index'))
+
+
+@auth.route('/confirm')
+@login_required
+def resend_confirm():
+    token = current_user.generate_token()
+    if current_app.config['DEBUG']:
+        print(url_for('auth.confirm', token=token, _external=True))
+    send_email(current_user.email, 'Confirm Your Account',
+               'auth/email/confirm', user=current_user, token=token)
+    flash('A new confirmation email is sent to your email address. \
+            You have 24 hours to verify your account.')
+    return redirect(url_for('main.index'))
+
+
+@auth.route('/unconfirmed')
```

Comment:
Remove this route as it is unnecessary

Reply:
since this is removed, I think we don't need "unconfirmed.html.j2"